src/components/updatemenus/draw.js

```diff
@@ -362,11 +417,51 @@ function findDimenstions(gd, menuOpts) {
             tLines = tspans[0].length || 1,
             hEff = Math.max(tHeight * tLines, constants.minHeight) + constants.textOffsetY;
 
-        menuOpts.width = Math.max(menuOpts.width, wEff);
+        hEff = Math.ceil(hEff);
+        wEff = Math.ceil(wEff);
+
+        // Store per-item sizes since a row of horizontal buttons, for example,
+        // don't all need to be the same width:
+        menuOpts.widths[i] = wEff;
```

Comment:
Interesting. Yeah, I had to do some carefully wacky stuff to get it to handle all the cases (h/v, buttons/dropdown, reverse/nonreverse, anchor=t/b/l/r—which is at least no less than like 2^5 = 32 cases), but I think it works…